aliyun-ecs/src/test/java/org/jclouds/aliyun/ecs/compute/internal/BaseECSComputeServiceApiMockTest.java

```diff
@@ -105,11 +105,22 @@ protected RecordedRequest assertSent(MockWebServer server, String method, String
       return request;
    }
 
-   protected RecordedRequest assertSent(MockWebServer server, String method, String action, Integer page) throws InterruptedException {
+   protected RecordedRequest assertSent(MockWebServer server, String method, String action, String location) throws InterruptedException {
       RecordedRequest request = server.takeRequest();
       assertEquals(request.getMethod(), method);
       Map<String, String> queryParameters = Splitter.on('&').trimResults().withKeyValueSeparator("=").split(request.getPath());
       assertEquals(queryParameters.get("Action"), action);
+      assertEquals(queryParameters.get("RegionId"), location);
+      assertEquals(request.getHeader("Accept"), "application/json");
+      return request;
+   }
```

Comment:
Instead of doing this in a method that should be generic, add a way to pass the query parameters. Otherwise you have a method in the base class that only works for one very concrete query parameter, which is not good.

Reply:
I think you are right, the 3 methods were a smell for me too. I'll come up with something better